feat(cash credit not) up to date

jeremielodi opened the pull request:
closes #3269

jeremielodi committed:
feat(cash credit not) up to date

jniles commented on client/src/modules/cash/modals/modal-cancel-cash.ctrl.js:
```diff
@@ -2,36 +2,58 @@ angular.module('bhima.controllers')
 
-    return Vouchers.reverse(vm.cancelCash)
-      .then(function () {
+    Vouchers.reverse(vm.cancelCash)
```
By removing the `return` of the promise, this breaks the loading indicator on the submit button.  It would be better to put this back and add a "0" to the other `return`.

jniles reviewed (commented):
@jeremielodi this addresses the majority of the points in #3269.  It also introduces a bug on the loading button which should be fixed.

We still need to show the invoices that were paid as requested in the issue.  Ideally, It should be the invoice that goes in the table, not the cash payment (we can only reverse a single cash payment at a time, so we'll never have more than one in a table).  This would be similar to the credit note, where the credit note shows each inventory item that the invoice was for.  In this case, we should show each invoice that the payment was for.  It was be nice to link them as well with something like `bhReferenceLink`.

Also, while you are here, could you remove the  `<div class="bhima-title">`?  It makes the modal head much larger than any of our other modals.

Great first step!

jniles committed:
fix(cash): remove entity identifier from find()

The entity identifier pattern can lead to incorrect project ID
assignments since it is sometimes computed based on the current
project instead of the patient's project.  We've removed that for
pulling from the document_map and entity_map.

jniles committed:
refactor(cash): improve layout of cash cancel

The cash cancel modal has been revamped with a layout more resembling
the credit note modal of the invoice page.  Additionally, the confusing
code that referenced "invoice" (since it was copied and pasted) has been
removed to properly reference cash payments when they need to be
reference and invoices when they need to be referenced.

Additionally, the case of cautions are handled, so that we do not
display a list of invoices in this case.

jniles committed:
fix(cash): poor variable name
